interview-questions.md

```diff
@@ -9,59 +9,59 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is `this` in JavaScript?
 
-  Your answer:
+  Your answer: 'this' is a continuation of the function so that it can be called upon later in the code
 
-  Researched answer:
+  Researched answer: This is to be able to reference the object that you are using within the Class.
 
 
 
 2. What is React? Why would you use it?
 
-  Your answer:
+  Your answer: React allows the user to be able to see their code in real time and also allows the user to interact with the code in real time without having to reload the page.
 
-  Researched answer:
+  Researched answer: React is a javascript library that enhances someones code by making it interactice and easy to use UI
 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
+  Your answer: This is something that i do no recall but I will research the answer and get it to you as soon as I can. 
 
-  Researched answer:
+  Researched answer: This is the method that is called when the component is mounted to the DOM.
 
 
 
 4. What is JSX? What is one syntax difference between HTML and JSX?
 
-  Your answer:
+  Your answer: JSX is a extension that allows HTMl to be used in javascript. One syntax thats different is the use of curly braces.
 
-  Researched answer:
+  Researched answer: JSX stands for Javascript XML and is a syntax extension to HTML. It allows for HTML to be used in Javascript and makes it easier for HTML to be used for react as well. 
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: Yarn is a file data base that after you install it allows you to test code in real time. I do not recall the file that is modified when you run yarn.
 
-  Researched answer:
+  Researched answer: Yarn is a package manager that guarantees stability if a code is working to continue to work in the future and is open sourced so that anyone can use it. The files that are modified when you run yarn are the package.json file and the package-lock.json file.
```

Comment:
Primarily, yarn will update the node modules according to the dependencies written in package.json